### DISCUSSION TITLE
Support Visual Studio Build Tools (16.9.4)

### DISCUSSION THREAD

**Discussion by ncook-hxgn** [wixtoolset/issues#6453] [Questions] [answered]
## Feature requests

If this issue is a feature request:

* Describe the scenario and benefits that the feature supports.

> wix extension can be used with Visual Studio 2019 16.9.4

* Describe how you're accomplishing the feature today (if possible).

> I'm not.

* Describe what you'd like the new feature to do.

> allow installation of the Wix Visual Studio Extension 2019 into Visual Studio 2019 16.9.4

**Comment by robmen** [comment, marked as answer]
I have Votive installed in VS2019 16.9.4 right now. We'd need more information to be of any use in helping you understand what the problem is with your configuration.

**Reply by ncook-hxgn** [reply]
Hi Rob,

Yes it seems I've been a bit dumb, this may turn into a dumb question*.

I am indeed able to install to VS 2019 16.9.4 Professional.. but not into the build tools. This is probably because there isn't an IDE I guess?

![image](https://user-images.githubusercontent.com/36889813/118104763-72c64580-b3d3-11eb-8e56-8bf5e3ba39eb.png)

* ..... Would it be possible for this extension to be installable into an IDE that I don't have installed? 

Context: This extension is a build dependency for one of our products, and we're trying to slim down our build servers by not installing the full fat IDE on this compiler upgrade round. Does our dependency on this extension mean we need to have the full IDE installed?

**Reply by wyrdfish** [reply]
The build tools are in the main installer, the visual studio extensions just add IDE integration.
is your pipeline failing with just the build tools installed? how are you calling the compilation?

**Reply by ncook-hxgn** [reply]
Ah OK - we were trying to work this out on our side. Thanks for confirming :)

We haven't actually run it yet, we didn't think it would work. I'll ask the team to spin it up and come back to you.. probably with good news.

**Reply by robmen** [reply]
To be specific, the Visual Studio Build Tools do not support extensions. This is a Visual Studio design limitation. Thus "WiX Toolset Visual Studio 2019 Extension" only provides the IDE integration.

You want the [WiX Toolset Build Tools](https://marketplace.visualstudio.com/items?itemName=WixToolset.WiXToolset) which provides the build tools... just like the Visual Studio Build Tools only provides build tools.

**Reply by ncook-hxgn** [reply]
Yes, I have several versions of the build tools already installed on several build agents for several compiler versions. I just don't develop the product so I didn't have initimate knowledge of what they all actually do ;)
